wagtail/wagtailusers/tests.py

```diff
@@ -348,6 +380,62 @@ def test_edit(self):
         # Check that the user was edited
         user = get_user_model().objects.get(pk=self.test_user.pk)
         self.assertEqual(user.first_name, 'Edited')
+        # Check that the user is no longer superuser
+        self.assertEqual(user.is_superuser, False)
+        # Check that the user is no longer active
+        self.assertEqual(user.is_active, False)
+
+
+    def test_edit_and_make_superuser(self):
+        response = self.post({
+            'username': "testuser",
+            'email': "test@user.com",
+            'first_name': "Edited",
+            'last_name': "User",
+            'password1': "password",
+            'password2': "password",
+            'is_active': 'on',
+            'is_superuser': 'on',
+        })
+
+        # Should redirect back to index
+        self.assertRedirects(response, reverse('wagtailusers_users:index'))
+
+        # Check that the user was edited
+        user = get_user_model().objects.get(pk=self.test_user.pk)
+
+        # Check that the user is now superuser
+        self.assertEqual(user.is_superuser, True)
+        # Check that the user is now active
+        self.assertEqual(user.is_active, True)
+
+    def test_edit_self(self):
+        response = self.post({
+            'username': 'test@email.com',
+            'email': 'test@email.com',
+            'first_name': "Edited Myself",
+            'last_name': "User",
+            # 'password1': "password",
+            # 'password2': "password",
+            'is_active': 'on',
+            'is_superuser': 'on',
+        }, self.current_user.pk)
+
+        # Should redirect back to index
+        self.assertRedirects(response, reverse('wagtailusers_users:index'))
+
+        # Check that the user was edited
+        user = get_user_model().objects.get(pk=self.current_user.pk)
+        self.assertEqual(user.first_name, 'Edited Myself')
+
+        # Check that the user is still superuser
+        self.assertEqual(user.is_superuser, True)
+        # Check that the user is still active
+        self.assertEqual(user.is_active, True)
+
+
+    def test_(arg):
```

Comment:
...actually, it looks like we're still missing a test to confirm that a superuser can't remove their own is_active / is_superuser flags, so perhaps that was intended to go here? :-)